test/pylib/coverage_utils.py

```diff
@@ -0,0 +1,1286 @@
+#!/usr/bin/python
+import argparse
+import asyncio
+import multiprocessing
+from pathlib import Path,PurePath
+from typing import Union,List,Dict,Iterable,Optional,Coroutine,Mapping,Any,List,Self
+import re
+from enum import Enum
+from asyncio.subprocess import PIPE,DEVNULL
+from collections.abc import Iterable as IterableType
+import os
+from collections import namedtuple
+from itertools import repeat
+from functools import wraps,partial
+import logging
+import sys
+import inspect
+#So the module can be imported outside of this directory
+sys.path.insert(0,os.path.join(os.path.dirname(__file__)))
```

Comment:
I don't understand how this enables importing this file?

Reply:
`coverage_utils` imports lcov_utils which resided in the same directory.
However, for example if you `import coverage_utils` in `test.py` without this "trick",
the import will fail because python will claim that it couldn't find lcov_utils module.
python default behavior for searching modules is:
1. `cwd` (which is the main source tree when running test.py)
2. The system defined directories

according to these rules, python will not search in `test/pylib` which is where lcov_utils resides. Adding this line tells python to also search in the local directory where `coverage_utils.py` resides. And then removes this rule immediately after finding the local module so following imports in the external file (which in this example is `test.py`) will not import by accident additional module from that directory that happen to have names identical to the system modules. (look at the 2 lines that comes right after this line)

Reply:
@denesb let me know if it answers you question.